### PR TITLE
Add a renderer for actions

### PR DESCRIPTION
**Pull request by masticore252**
This adds customization support to the action buttons

this can help with #19 until proper localization support is implemented

### CODE REVIEW

File: src/actions/default.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    canDeleteFile,
+    onDeleteFile,
+
+    canDeleteFolder,
+    onDeleteFolder,
+
+    canDownloadFile,
+    onDownloadFile,
+
+  } = props
```
[the remark this answers is not in the document]
Comment (masticore252) [reply src/actions/default.js]: I added PropTypes checks on a new commit, please review it and let me know if you see something wrong

File: src/browser.js [hunk 2]
```diff
@@ -488,135 +495,36 @@ class RawFileBrowser extends React.Component {
...
+
+        canRenameFile={typeof onRenameFile === 'function'}
+        onRenameFile={this.handleActionBarRenameClick}
+
+        canRenameFolder={typeof onRenameFolder === 'function'}
+        onRenameFolder={this.handleActionBarRenameClick}
+
+        canDeleteFile={typeof onDeleteFile   === 'function'}
+        onDeleteFile={this.handleActionBarDeleteClick}
+
```
[the remark this answers is not in the document]
Comment (masticore252) [reply src/browser.js]: fixed

File: src/browser.js [hunk 3]
```diff
@@ -488,135 +495,36 @@ class RawFileBrowser extends React.Component {
...
-      if (actions.length) {
-        actions = (<ul className="item-actions">{actions}</ul>)
-      } else {
-        actions = (<div className="item-actions">&nbsp;</div>)
-      }
-    }
+    let actions = null
+    actions = (
+      <ActionRenderer
+        {...actionRendererProps}
```
[the remark this answers is not in the document]
Comment (masticore252) [reply src/browser.js]: you're right, I 've removed them

File: src/actions/default.js [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+          </a>
+        </li>
+      )
+    }
+
+    if (actions.length) {
+      actions = (<ul className="item-actions">{actions}</ul>)
+    } else {
+      actions = (<div className="item-actions">&nbsp;</div>)
+    }
```
Comment (scaredcat) [inline src/actions/default.js]: this could be moved right before returning `actions` which means you could delete the other identical block.

**Pull request merged**